thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datasource/sql/SqlThirdEyeDataSource.java

```diff
@@ -0,0 +1,204 @@
+  }
+
+  @Override
+  public void clear() throws Exception {
+    // left blank
+  }
+
+  @Override
+  public void close() throws Exception {
+    // left blank
+  }
+
+  @Override
+  public long getMaxDataTime(String dataset) throws Exception {
+    LOG.info("Getting max data time for " + dataset);
+    return sqlResponseCacheLoader.getMaxDataTime(dataset);
+  }
+
+  @Override
+  public Map<String, List<String>> getDimensionFilters(String dataset) throws Exception {
+    LOG.info("Running dimension filters for " + dataset);
+    return this.sqlResponseCacheLoader.getDimensionFilters(dataset);
+  }
+
+
+  private static LoadingCache<SqlQuery, ThirdEyeResultSet> buildResponseCache(
+      SqlResponseCacheLoader sqlResponseCacheLoader) throws Exception {
+    Preconditions.checkNotNull(sqlResponseCacheLoader, "A loader that sends query to Pinot is required.");
+
+    // Initializes listener that prints expired entries in debuggin mode.
+    RemovalListener<SqlQuery, ThirdEyeResultSet> listener;
+    if (LOG.isDebugEnabled()) {
+      listener = new RemovalListener<SqlQuery, ThirdEyeResultSet>() {
+        @Override
+        public void onRemoval(RemovalNotification<SqlQuery, ThirdEyeResultSet> notification) {
+          LOG.debug("Expired {}", notification.getKey().getSql());
+        }
+      };
+    } else {
+      listener = new RemovalListener<SqlQuery, ThirdEyeResultSet>() {
+        @Override public void onRemoval(RemovalNotification<SqlQuery, ThirdEyeResultSet> notification) { }
+      };
+    }
+
+    // ResultSetGroup Cache. The size of this cache is limited by the total number of buckets in all ResultSetGroup.
+    // We estimate that 1 bucket (including overhead) consumes 1KB and this cache is allowed to use up to 50% of max
+    // heap space.
+    long maxBucketNumber = getApproximateMaxBucketNumber(DEFAULT_HEAP_PERCENTAGE_FOR_RESULTSETGROUP_CACHE);
+    LOG.debug("Max bucket number for {}'s cache is set to {}", DATA_SOURCE_NAME, maxBucketNumber);
+
+    return CacheBuilder.newBuilder()
+        .removalListener(listener)
+        .expireAfterWrite(15, TimeUnit.MINUTES)
+        .maximumWeight(maxBucketNumber)
+        .weigher(new Weigher<SqlQuery, ThirdEyeResultSet>() {
+          @Override public int weigh(SqlQuery SQLQuery, ThirdEyeResultSet resultSet) {
+              return ((resultSet.getColumnCount() + resultSet.getGroupKeyLength()) * resultSet.getRowCount());
+          }
+        })
+        .build(sqlResponseCacheLoader);
+  }
+
+  private static long getApproximateMaxBucketNumber(int percentage) {
+    long jvmMaxMemoryInBytes = Runtime.getRuntime().maxMemory();
+    if (jvmMaxMemoryInBytes == Long.MAX_VALUE) { // Check upper bound
+      jvmMaxMemoryInBytes = DEFAULT_UPPER_BOUND_OF_RESULTSETGROUP_CACHE_SIZE_IN_MB * 1048576L; // MB to Bytes
+    } else { // Check lower bound
+      long lowerBoundInBytes = DEFAULT_LOWER_BOUND_OF_RESULTSETGROUP_CACHE_SIZE_IN_MB * 1048576L; // MB to Bytes
```

Comment:
You can use some library to refer to standard constants.
For example, `FileUtils.ONE_MB`

http://commons.apache.org/proper/commons-io//javadocs/api-2.2/org/apache/commons/io/FileUtils.html